main.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python
+
+import os
+from datetime import datetime, timezone, timedelta
+from requests import get, post
+import threading
+import logging
+import re
+import json
+
+# semaphore limit of 5, picked this number arbitrarily
+maxthreads = 5
+sema = threading.Semaphore(value=maxthreads)
+
+logging.basicConfig(level=logging.DEBUG)
+logger = logging.getLogger()
+
+
+PAGERDUTY_API_KEY = os.environ['PAGERDUTY_API_KEY']
+SLACK_API_KEY = os.environ['SLACK_API_KEY']
+# [{"slack_channel_id": "foo", "pd_schedule_id": "bar"},{"slack_channel_id": "boo", "pd_schedule_id": "baz,moz"}]
+SCHEDULE_CONFIG = os.environ['SCHEDULE_CONFIG']
+
+
+def get_user(schedule_id):
+    headers = {
+        'Accept': 'application/vnd.pagerduty+json;version=2',
+        'Authorization': f"Token token={PAGERDUTY_API_KEY}"
```

Comment:
TIL about f-Strings:

https://realpython.com/python-f-strings/#f-strings-a-new-and-improved-way-to-format-strings-in-python